edit/workload/storage/PVC.vue

```diff
@@ -0,0 +1,96 @@
+<script>
+import VolumeMount from '@/edit/workload/storage/volume-mount.js';
+import Mount from '@/edit/workload/storage/Mount';
+import LabeledInput from '@/components/form/LabeledInput';
+import LabeledSelect from '@/components/form/LabeledSelect';
+import Checkbox from '@/components/form/Checkbox';
+import { mapGetters } from 'vuex';
+import PersistentVolumeClaim from '@/edit/PersistentVolumeClaim';
+import { PVC } from '@/config/types';
+
+export default {
+  components: {
+    LabeledInput,
+    LabeledSelect,
+    Mount,
+    PersistentVolumeClaim,
+    Checkbox
+  },
+  mixins: [VolumeMount],
+
+  props:      {
+    mode: {
+      type:    String,
+      default: 'create'
+    },
+
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+
+    // array of existing persistent volume claims
+    pvcs: {
+      type:    Array,
+      default: () => []
+    },
+
+    registerBeforeHook: {
+      type:    Function,
+      default: null,
+    },
+  },
+
+  async fetch() {
+    // TODO get namespace for this deployment
+    const namespace = this.$store.getters['defaultNamespace'];
```

Comment:
Yes it will have to match the workload being created, including handling if you click add PVC and then later change the workload namespace before clicking save.